[quant][graphmode][fx] Enable inception_v3 googlenet and mobilenet_v2 static quant test

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#51402 [quant][graphmode][fx] Enable inception_v3 googlenet and mobilenet_v2 static quant test**

Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D26162805](https://our.internmc.facebook.com/intern/diff/D26162805)

jerryzh168 committed:
[quant][graphmode][fx] Enable inception_v3 and googlenet static quant test

Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fx] Enable inception_v3 and googlenet static quant test"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D26162805](https://our.internmc.facebook.com/intern/diff/D26162805)

[ghstack-poisoned]

vkuzo commented on test/quantization/test_quantize_fx.py:
```diff
@@ -2839,6 +2838,8 @@ def _test_model_impl(
         # print('quantized and scripted:', qgraph_script.graph)
 
+        import copy
+        qgraph_input = copy.deepcopy(input_value)
```
just for my own curiousity, what breaks without the copy?

jerryzh168 replied:
I think we can remove it, it's added when i‘m debugging

vkuzo reviewed (approved):
awesome!

raghuramank100 commented on test/quantization/test_quantize_fx.py:
```diff
@@ -2962,31 +2965,46 @@ def get_available_classification_models(models):
+            ("mobilenet_v2", "qat"),
+            ("inception_v3", "qat"),
+            ("googlenet", "qat")
+        ]  # because relu6 is replaced as relu in mobilenetv2
```
Cant you replace relu6 with relu by a separate pass and match with eager for mobilenetv2?

raghuramank100 commented on test/quantization/test_quantize_fx.py:
```diff
@@ -2962,31 +2965,46 @@ def get_available_classification_models(models):
             pretrained = name in quantized_model_list  # load pretrained model to compare with quantized model
+            kwargs = {}
+            # turn off transform input for inception_v3 since
+            # it's not quantized in eager mode and in fx graph
```
Also, it is a good idea to update the test in torchvision for quantization at: https://github.com/pytorch/vision/blob/master/test/test_quantized_models.py#L31

jerryzh168 replied:
sure, will do it later

raghuramank100 commented on test/quantization/test_quantize_fx.py:
```diff
@@ -2962,31 +2965,46 @@ def get_available_classification_models(models):
-            check_with_eager = name not in fx_eager_not_matching
+            model = models.__dict__[name](pretrained=pretrained, **kwargs).eval().float()
+            if name == "mobilenet_v2":
+                _replace_relu(model)
```
Since you are replacing relu, why does mobilenetv2 not match?

jerryzh168 replied:
it matches in ptq static quantization, but failed in qat, because of AdaptiveAveragePool

raghuramank100 commented on test/quantization/test_quantize_fx.py:
```diff
@@ -2881,6 +2882,7 @@ def _test_model_impl(
             qeager.eval()
 
             # print('ref after quantization:', qeager)
+            qeager_input = copy.deepcopy(input_value)
```
Is this used anywhere else?

jerryzh168 replied:
no, will remove, it's added when debugging

jerryzh168 committed:
Update on "[quant][graphmode][fx] Enable inception_v3 googlenet and mobilenet_v2 static quant test"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D26162805](https://our.internmc.facebook.com/intern/diff/D26162805)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fx] Enable inception_v3 googlenet and mobilenet_v2 static quant test"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D26162805](https://our.internmc.facebook.com/intern/diff/D26162805)

[ghstack-poisoned]
